Project structure and DB

Dolgoff62 opened the pull request.

Dolgoff62 committed:
added folders structure

Created a project structure (folders), added data classes: User, patient, note.

Dolgoff62 committed:
added empty file

added empty file ".gitkeep" to each folder

Dolgoff62 committed:
Add Db and Entitys

Added a database and tables for Users, patients and notes.

Dolgoff62 committed:
Project optimization

Added coroutine dependency. Removed unnecessary function from entity

Dolgoff62 committed:
Added delete function in dao

Added functions for deleting entities: patient, user, note from the database

Dolgoff62 committed:
Fix

Renaming of class date fields has been performed.
Changed property types for date classes.

Dolgoff62 committed:
Merge branch 'Project-structure' of https://github.com/Comdcomp/FMHAndroid into Project-structure

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/dao/NoteDao.kt:
```diff
@@ -0,0 +1,24 @@
+    @Insert(onConflict = OnConflictStrategy.REPLACE)
+    suspend fun insert(notes: List<NoteEntity>)
+
+    @Query("UPDATE NoteEntity Set deleted = 1 WHERE id = :id")
```
Для всех остальных запросов к базе не надо фильтровать результат по полю `deleted`? Или это в каком-нибудь перехватчике запросов прописано?

Dolgoff62 replied:
Пока под вопросом.

Dolgoff62 replied:
теперь такая необходимость отпала, в связи с изменением бэка..

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/dao/PatientDao.kt:
```diff
@@ -0,0 +1,26 @@
+    @Insert(onConflict = OnConflictStrategy.REPLACE)
+    suspend fun insert(patients: List<PatientEntity>)
+
+    @Query("UPDATE PatientEntity SET deleted = 1 WHERE id = :id")
```
Для всех остальных запросов к базе не надо фильтровать результат по полю `deleted`? Или это в каком-нибудь перехватчике запросов прописано?

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/dao/UserDao.kt:
```diff
@@ -0,0 +1,27 @@
+    @Insert(onConflict = OnConflictStrategy.REPLACE)
+    suspend fun insert(users: List<UserEntity>)
+
+    @Query("UPDATE UserEntity SET deleted = 1 WHERE id = :id")
```
Для всех остальных запросов к базе не надо фильтровать результат по полю `deleted`? Или это в каком-нибудь перехватчике запросов прописано?

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/entity/NoteEntity.kt:
```diff
@@ -0,0 +1,55 @@
+@Entity
+data class NoteEntity(
+    @PrimaryKey
+    val id: Int,
+    val patientId: Int,
+    var description: String,
+    val creatorId: Int,
+    val executorId: Int,
+    val createDate: String,
+    val planeExecuteDate: String,
+    val factExecuteDate: String,
+    val statusId: Int,
+    var comment: String,
+    val deleted: Boolean
```
Не зафиксированы названия полей таблицы. Проверяли переживёт ли код обфускацию? Релизная сборка по умолчанию обфусцирует код. Если прописан `proguard`-файл, то хорошо. Но надёжнее в итоге будет прописать ручками названия полей таблицы и убрать из `proguard`-файла ограничение на обфускацию сущностей базы данных

Dolgoff62 replied:
Исправлено

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/entity/NoteEntity.kt:
```diff
@@ -0,0 +1,55 @@
+    val factExecuteDate: String,
+    val statusId: Int,
+    var comment: String,
+    val deleted: Boolean
```
Похоже, что поля `id` и`deleted` будут у всех сущностей в базе данных. Может тогда вынести их в общий родительский класс?

Dolgoff62 replied:
А как же: (до версии 1.1) Классы данных не могут наследоваться от других классов (но могут реализовывать интерфейсы)?

Dolgoff62 replied:
Оставлено в изначальном варианте.

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/entity/NoteEntity.kt:
```diff
@@ -0,0 +1,55 @@
+    )
+
+    companion object {
+        fun fromDto(dto: Note) = NoteEntity(
```
Почему бы, кстати, не конвертнуть в `extension`-функцию и переименовать в `toEntity`? Получилось бы меньше кода

Dolgoff62 replied:
Исправлено

maks1596 reviewed (commented):
Не стал дублировать комментарии для различных сущностей

Dolgoff62 committed:
fix

Dolgoff62 committed:
Multiple changes

Changed the structure to single activities and fragments. Added feed snippet. Added navigation by fragments. Additional entities have been created. The repository for the patient is registered.

Dolgoff62 commented on app/src/main/java/ru/netology/fmhandroid/api/PatientApiService.kt:
```diff
@@ -32,6 +32,9 @@ private val retrofit = Retrofit.Builder()
     .build()
 
 interface PatientApiService {
```
Тут конечно вопрос. Я здесь жестко прописал в запросе количество пациентов (0-30) и всех (false). Это конечно не правильно, но мы не знаем сколько сейчас всего пациентов и каков максимум. Почему 30? Потому что в общей группе в разговоре ребята упоминали, что пока вроде бы планируется максимально 30 коек в хосписе. 
Но все равно меня этот запрос сильно смущает..

maks1596 replied:
Лучше вынести наружу эту параметризацию, а не забивать прямо в запрос. А вот снаружи уже пускай всегда передаются константы.
Я в данном случае рассуждаю так, что это у нас API сервера, которое надо менять, только когда API сервера действительно поменяется, а не когда мы решим получать 15 пациентов на страницу

maks1596 replied:
И зачем вообще константы? 30 - это не так много. давай просто запросим все, коли у нас всё равно пагинации нет и о количестве думать не приходится

Dolgoff62 replied:
Ждем бекэнд

Dolgoff62 replied:
Так же проблема разрешилась с изменением сервера. Теперь цепляем всех пользователей по статусу)

maks1596 commented on app/build.gradle:
```diff
@@ -20,7 +21,13 @@ android {
             minifyEnabled false
+            manifestPlaceholders.usesCleartextTraffic = false
             proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
+            buildConfigField "String", "BASE_URL", '"http://130.193.44.96:8080/fmh/"'
```
Надо погуглить. Как-то можно для всех типов сборки сразу прописать одну и ту же переменную, чтобы не дублировать этот код для `debug`. Но скорей всего адреса со временем станут разными, поэтому нет большого смысла

Dolgoff62 replied:
Погуглю обязательно.
Пока оставлено в исходном варианте.

maks1596 commented on app/build.gradle:
```diff
@@ -37,13 +44,32 @@ android {
+    implementation "com.squareup.okhttp3:logging-interceptor:$okhttploggin_version"
+
+
+    testImplementation "androidx.room:room-testing:$room_version"
```
У нас же нет тестов в проекте. Зачем для них зависимости подтягиваются? 😊

Dolgoff62 replied:
Логично. Как всегда на будущее делал, но как говорится - лишний код "под нож" :)
Исправлено.

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/api/PatientApiService.kt:
```diff
@@ -0,0 +1,58 @@
+import java.util.concurrent.TimeUnit
+
+
+private val loggin = HttpLoggingInterceptor().apply {
+    if (BuildConfig.DEBUG) {
+        level = HttpLoggingInterceptor.Level.BODY
+    }
+}
+
+private val okhttp = OkHttpClient.Builder()
+    .addInterceptor(loggin)
+    .connectTimeout(30, TimeUnit.SECONDS)
+    .build()
+
+private val retrofit = Retrofit.Builder()
+    .addConverterFactory(GsonConverterFactory.create())
+    .baseUrl(BASE_URL)
+    .client(okhttp)
+    .build()
```
Занести в `PatientApi`, иначе эти объекты будут создаваться раньше времени. `object` же в котлине (коим является `PatientApi`) создастся только при первом обращении к нему

Dolgoff62 replied:
Исправлено.

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/api/PatientApiService.kt:
```diff
@@ -0,0 +1,58 @@
+
+private val loggin = HttpLoggingInterceptor().apply {
+    if (BuildConfig.DEBUG) {
+        level = HttpLoggingInterceptor.Level.BODY
```
Если не дебаг, то `level` принимает неизвестно какое значение 😊
Оптимальнее, надёжнее и понятнее будет впринципе не создавать и не добавлять перехватчик `loggin`, если только у нас не отладочный режим

Dolgoff62 replied:
Исправлено.

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/api/PatientApiService.kt:
```diff
@@ -0,0 +1,58 @@
+import java.util.concurrent.TimeUnit
+
+
+private val loggin = HttpLoggingInterceptor().apply {
```
По-моему опечатка. Вроде как не хватает `...g` в конце названия переменной

Dolgoff62 replied:
Да, была опечатка.
Исправлено.

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/api/PatientApiService.kt:
```diff
@@ -0,0 +1,58 @@
+    @PATCH("patient")
+    suspend fun editPatient(@Body patient: Patient): Response<Patient>
+
+    object PatientApi {
```
```suggestion
    companion object {
```
И вуаля - тебе не надо запоминать ещё одно имя класса (и придумывать его, что ещё страшнее 🙀)

Dolgoff62 replied:
Хм. И как я сам не догадался, отличный вариант, везде где мы обращаемся к Api Service теперь код меньше и красивее)))
Исправлено.

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/api/PatientApiService.kt:
```diff
@@ -0,0 +1,58 @@
+    .client(okhttp)
+    .build()
+
+interface PatientApiService {
```
Может оставить в имени просто `PatientApi`? Суффикс `...Service` я бы предпочёл добавлять к андроидовским сервисам

Dolgoff62 replied:
Согласен.
Исправлено.

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/dto/Admission.kt:
```diff
@@ -0,0 +1,11 @@
+package ru.netology.fmhandroid.dto
+
+data class Admission(
+    val id: Int,
+    val dateIn: String,
+    val dateOut: String,
+    val factIn: Boolean = false,
+    val factOut: Boolean = false,
+    val deleted: Boolean = false,
+    val patientId: Int
+)
```
Задокументировать. У меня прям падает подозрение на поля `factIn` и `factOut`.
Для документации использовать формат KDoc. Для упрощения жизни поставить плагин любой понравившийся плагин в Android Studio, который поможет формировать `KDoc`-комментарии. 

P.S.: Советую комментарии прописывать над каждым свойством, а не через `@property`

Dolgoff62 replied:
Исправлено.

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/dto/Patient.kt:
```diff
@@ -0,0 +1,12 @@
+    var lastName: String,
+    var middleName: String,
+    val birthDate: String,
+    val shortPatientName: String,
```
Что это за поле? 😊
Готов поспорить, что оно имеет смысл только для слоя отображения, но никак не для базы данных, бизнес-логики или передачи данных на сервер

Dolgoff62 replied:
Точно так.
Исправлено.

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/entity/AdmissionEntity.kt:
```diff
@@ -0,0 +1,47 @@
+import androidx.room.PrimaryKey
+import ru.netology.fmhandroid.dto.Admission
+
+@Entity(tableName = "AdmissionEntity")
+data class AdmissionEntity(
+    @PrimaryKey
+    @ColumnInfo(name = "id")
+    val id: Int,
+    @ColumnInfo(name = "dateIn")
+    val dateIn: String,
+    @ColumnInfo(name = "dateOut")
+    val dateOut: String,
+    @ColumnInfo(name = "factIn")
+    val factIn: Boolean = false,
+    @ColumnInfo(name = "factOut")
+    val factOut: Boolean = false,
+    @ColumnInfo(name = "deleted")
+    val deleted: Boolean = false,
+    @ColumnInfo(name = "patientId")
+    val patientId: Int
+) {
```
Имена таблиц и полей лучше взять из ER-диаграммы. Там выбран хороший классический стиль именования для БД

Dolgoff62 replied:
Исправлено

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/entity/AdmissionEntity.kt:
```diff
@@ -0,0 +1,47 @@
+    @ColumnInfo(name = "patientId")
+    val patientId: Int
+) {
+    fun toDto() = Admission(
+        id,
+        dateIn,
+        dateOut,
+        factIn,
+        factOut,
+        deleted,
+        patientId
+    )
```
Добавить имена параметров. 
В таком варианте никто не заметит ошибки, если перепутать местами `dateIn` и `dateOut` или `factIn` и `factOut`.

P.S.: Поставить каретку ввода на конструктор -> `Alt+Enter` -> `Add names to parameters`

Dolgoff62 replied:
Очень удобно. Спасибо. Буду пользоваться.
Исправлено.

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/repository/patientRepository/PatientRepository.kt:
```diff
@@ -0,0 +1,15 @@
+import ru.netology.fmhandroid.dto.Patient
+
+interface PatientRepository {
+    val data: Flow<List<Patient>>
+    suspend fun getAllPatients()
```
Наверное надо схлопнуть вместе - очень странно, что `get`-метод ничего не возвращает

Dolgoff62 replied:
метод getAll() нужен для закачки списка пациентов с сервера и "запихивания" его в локальную БД.
Поэтому он вынесен отдельно. Так как к data мы будем обращаться очень часто, а запрос на сервер так же часто нам не требуется.

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/repository/patientRepository/PatientRepositoryImp.kt:
```diff
@@ -0,0 +1,120 @@
+            throw ServerError
+        } catch (e: Exception) {
+            throw UnknownError
+        }
```
У тебя этот паттерн повторяется в каждой функции репозитория. Вынеси всё общее в отдельную приватную функцию. Используй паттерн `шаблонный метод` за счёт великолепных лямбд котлина

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/repository/patientRepository/PatientRepositoryImp.kt:
```diff
@@ -0,0 +1,120 @@
+                throw ApiError(response.code(), response.message())
+            }
+
+            val body = response.body() ?: throw ApiError(response.code(), response.message())
```
Лучше переименуй `body` в `patient`. `body` - это зачастую просто строка и по итогу странно видеть, как ты без десериализации возвращаешь `body` из функции, которая возвращает `Patient`

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/error/AppError.kt:
```diff
@@ -0,0 +1,20 @@
+import android.database.SQLException
+import java.io.IOException
+
+sealed class AppError(var code: String) : RuntimeException() {
```
```suggestion
sealed class AppException(var code: String) : RuntimeException() {
```
Обычно всё, что можно выбросить через `throw` именуют `Exception`. `Error` же можно безопасно вернуть через `return` из функции в качестве показателя неудачного выполнения функции, но нельзя выбросить через `throw`

P.S,: К наследникам тоже относится 😊

Dolgoff62 replied:
Исправлено.

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/repository/patientRepository/PatientRepositoryImp.kt:
```diff
@@ -0,0 +1,120 @@
+    override suspend fun savePatient(patient: Patient) {
+        try {
+            if (patient.id == 0) {
+                val response = PatientApiService.PatientApi.service.savePatient(patient)
+                if (!response.isSuccessful) {
+                    throw ApiError(response.code(), response.message())
+                }
+                val body = response.body() ?: throw ApiError(response.code(), response.message())
+                patientDao.insert(body.toEntity())
```
Каждая ветка `if`-а будет хорошо смотреться в собственной отдельной функции. Код станет прозрачнее и более читаемый.
> Если идентификатор пациента равен 0, то сохраняем нового пациент. Иначе, коли пациент уже есть, обновляем по нему данные

Dolgoff62 replied:
Исправлено

Dolgoff62 replied:
разнес по разным функциям.
Исправлено

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/ui/AppActivity.kt:
```diff
@@ -0,0 +1,25 @@
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        intent?.let {
```
```suggestion
        val intent = intent ?: return
```
И у тебя сразу уйдёт лишняя вложенность от `let`, код сместится влево и будет занимать меньше места

P.S. У `if`-ов ниже можно тоже убрать блочные скобки: `if (...) return`

Dolgoff62 replied:
Отлично )
Исправлено.

Dolgoff62 replied:
Исправлено

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/ui/AppActivity.kt:
```diff
@@ -0,0 +1,25 @@
+                return@let
+            }
+
+            val text = it.getStringExtra(Intent.EXTRA_TEXT)
```
Объясни, пожалуйста, что здесь происходит. Я не понимаю зачем ты проверяешь `EXTRA_TEXT` на то, что он не пустой

Dolgoff62 replied:
Как всегда забегал я вперед. Это было сделано для возможности отправки сообщений из приложения, проверка на то, что текст сообщения не пустой. 
Убрал это действие до возникновения необходимости.
Исправлено.

Dolgoff62 replied:
Это я на будущее делал, если надо будет оправлять сообщения в интернет, проверка на то что текст не пустой. 
Убрал пока этот код.

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/ui/FeedFragment.kt:
```diff
@@ -0,0 +1,31 @@
+            false
+        )
+
+        TODO("Not yet implemented")
```
Воу. Вылетит же приложение. Ты его не запускаешь? Может не надо так с порога валить приложение? Пусть хоть пустой экран будет показан

Dolgoff62 replied:
Исправлено.

Dolgoff62 committed:
some changes

Dolgoff62 committed:
try to refactoring fun in repo

Dolgoff62 committed:
work with repo func

Dolgoff62 committed:
code correction

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/dto/Admission.kt:
```diff
@@ -0,0 +1,32 @@
+package ru.netology.fmhandroid.dto
+
+/**
+ * Госпитацизация
```
Опечатка

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/dto/Admission.kt:
```diff
@@ -0,0 +1,32 @@
+     */
+    val planDateOut: String,
+    /**
+     * Вактическая дата госпитализации
```
опечатка

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/dto/Admission.kt:
```diff
@@ -0,0 +1,32 @@
+    /**
+     * Актуальный статус госпитализации
+     */
+    val admStatusId: Int,
```
Что такое "adm"? Не надо жалеть буквы - они бесплатные 😊

maks1596 replied:
А. Это наверное от "admission". Тогда просто не надо "adm", потому что мы и без того находимся в классе `Admission` и говорим про статус этой же сущности

Dolgoff62 committed:
naming fix

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/dto/Admission.kt:
```diff
@@ -0,0 +1,32 @@
+    /**
+     * Актуальный статус госпитализации
+     */
+    val admStatusId: Int,
```
Это статус госпитализации, как написано в комментарии, или всё же идентификатор статуса, как прописано кодом?

Dolgoff62 replied:
Идентификатор.

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/enum/PatientStatusList.kt:
```diff
@@ -0,0 +1,7 @@
+package ru.netology.fmhandroid.enum
```
Наверное лучше положить в пакет dto. А ещё лучше вложить в класс `Patient` и переименовать в `Status`

Dolgoff62 replied:
Исправлено

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/enum/PatientStatusList.kt:
```diff
@@ -0,0 +1,7 @@
+enum class PatientStatusList {
+    ACTIVE,
+    DISCHARGED,
+    EXPECTED
```
Сервер их немного иначе расшифровывает в отличии от буквального перевода - лучше задокументировать каждый статус

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/repository/patientRepository/PatientRepositoryImp.kt:
```diff
@@ -0,0 +1,113 @@
+import ru.netology.fmhandroid.repository.patientRepository.PatientRepository
+
+
+class PatientRepositoryImp(
```
`...Impl` - не `Imp` 🙂

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/repository/patientRepository/PatientRepositoryImp.kt:
```diff
@@ -0,0 +1,113 @@
+        request = { PatientApi.service.getAllPatients() },
+        onSuccess = { body ->
+            patientDao.insert(body.map { it.toEntity() })
+            body
```
У тебя везде `body` возвращается из лямбды. Думаю, в таком случае выгоднее переделать функцию `makeRequest`, чтобы она всегда самостоятельно возвращала `body`.
P.S.: Останется один generic-тип `T`

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/ui/FeedFragment.kt:
```diff
@@ -0,0 +1,31 @@
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+
+        val binding = FragmentFeedBinding.inflate(
+            inflater,
+            container,
+            false
+        )
+
+        return binding.root
+    }
```
Подправлять не обязательно, но можешь взять на карандаш следующий вариант:
```suggestion
    ) = FragmentFeedBinding.inflate(
        inflater,
        container,
        false
    ).root
```
Если появится код конфигурации, то всегда можно добавить `apply` или `also`:
```suggestion
    ) = FragmentFeedBinding.inflate(
        inflater,
        container,
        false
    ).also { binding ->
        ...
    }.root
```

Dolgoff62 replied:
Круто :) Записал себе в заметки этот вариант

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/repository/patientRepository/PatientRepository.kt:
```diff
@@ -0,0 +1,18 @@
+interface PatientRepository {
+    val data: Flow<List<Patient>>
+    suspend fun getAllPatients() : List<Patient>
+    suspend fun getAllPatientsWithAdmissionStatus(status: PatientStatusList) : List<Patient>
```
Это один метод на самом деле с опциональным полем для фильтрации по статусу

Dolgoff62 replied:
С этим я еще буду думать. Когда сделаем верстку для списка пациентов Эти методы изменим)

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/repository/patientRepository/PatientRepository.kt:
```diff
@@ -0,0 +1,18 @@
+import ru.netology.fmhandroid.dto.Patient
+import ru.netology.fmhandroid.enum.PatientStatusList
+
+interface PatientRepository {
```
Я бы объединил `data` и `getAllPatients` в один метод, который возвращает `Flow`. Иначе у тебя сейчас получается раскрытие деталей, что `data` берётся из БД, а `getAllPatients` - результат запроса к серверу (который, кстати, всё равно обновляет БД и стригерит обновление `data`). 
Попробуй задокументировать эти поля. Если получится, не раскрывая деталей (сеть и бд - это детали), сформулировать их отличия, то значит ты всё правильно сделал. Если нет, то надо объединять

Dolgoff62 replied:
Вот с этим возникают вопросы. Т.к. если я объединю эти два метода, то data придется делать suspend, что затруднит обращение к этой переменной из вьюмодели..

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/api/PatientApi.kt:
```diff
@@ -0,0 +1,56 @@
+
+
+interface PatientApi {
+    @GET("patient?patients_status_list=ACTIVE")
+    suspend fun getAllPatients(): Response<List<Patient>>
+
+    @GET("patient")
+    suspend fun getAllPatientsWithAdmissionStatus(
+        @Query("patients_status_list") status: PatientStatusEnum
+    ): Response<List<Patient>>
```
А нельзя на что-то подобное заменить?
```suggestion
    @GET("patient")
    suspend fun getAllPatients(
        @Query("patients_status_list") status: PatientStatusEnum = PatientStatusEnum.ACTIVE
    ): Response<List<Patient>>
```
Если здесь нельзя указать значение параметра по умолчанию, то вынеси этот параметр по умолчанию выше - на уровень репозитория

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/enum/PatientStatusEnum.kt:
```diff
@@ -0,0 +1,7 @@
+package ru.netology.fmhandroid.enum
+
+enum class PatientStatusEnum {
```
Просто `PatientStatus` 😊
Никогда не добавляются детали к названию класса. Важна семантика, а не детали реализации.
Я бы ещё этот класс занёс внутрь класса `Patient` и оставил просто `Status` - так и писать меньше, и лучше видна принадлежность одного класса к другому

Dolgoff62 committed:
combining data and getAllPatients into one method.

and a few small changes and fixes

Dolgoff62 committed:
fix
